spatial_cell_gpu.cpp

```diff
@@ -0,0 +1,1170 @@
+      list_with_replace_new = ::new (buf11) split::SplitVector<vmesh::GlobalID>(INIT_VMESH_SIZE*acc_reserve_multiplier);
+      list_delete = ::new (buf12) split::SplitVector<Hashinator::hash_pair<vmesh::GlobalID,vmesh::LocalID>>(INIT_VMESH_SIZE);
+      list_to_replace = ::new (buf13) split::SplitVector<Hashinator::hash_pair<vmesh::GlobalID,vmesh::LocalID>>(INIT_VMESH_SIZE);
+      list_with_replace_old = ::new (buf14) split::SplitVector<Hashinator::hash_pair<vmesh::GlobalID,vmesh::LocalID>>(INIT_VMESH_SIZE);
+      // list_with_replace_new = new split::SplitVector<vmesh::GlobalID>(INIT_VMESH_SIZE*acc_reserve_multiplier);
+      // list_delete = new split::SplitVector<Hashinator::hash_pair<vmesh::GlobalID,vmesh::LocalID>>(INIT_VMESH_SIZE);
+      // list_to_replace = new split::SplitVector<Hashinator::hash_pair<vmesh::GlobalID,vmesh::LocalID>>(INIT_VMESH_SIZE);
+      // list_with_replace_old = new split::SplitVector<Hashinator::hash_pair<vmesh::GlobalID,vmesh::LocalID>>(INIT_VMESH_SIZE);
+      dev_list_with_replace_new = list_with_replace_new->upload<true>();
+      dev_list_delete = list_delete->upload<true>();
+      dev_list_to_replace = list_to_replace->upload<true>();
+      dev_list_with_replace_old = list_with_replace_old->upload<true>();
+      list_with_replace_new_capacity = INIT_VMESH_SIZE*acc_reserve_multiplier;
+      list_delete_capacity = INIT_VMESH_SIZE;
+      list_to_replace_capacity = INIT_VMESH_SIZE;
+      list_with_replace_old_capacity = INIT_VMESH_SIZE;
+   }
+
+   SpatialCell::~SpatialCell() {
+      if (velocity_block_with_content_list) {
+         ::delete velocity_block_with_content_list;
+         velocity_block_with_content_list = 0;
+      }
+      if (velocity_block_with_content_map) {
+         ::delete velocity_block_with_content_map;
+         velocity_block_with_content_map = 0;
+      }
+      if (velocity_block_with_no_content_map) {
+         ::delete velocity_block_with_no_content_map;
+         velocity_block_with_no_content_map = 0;
+      }
+      velocity_block_with_content_list_size=0;
+      velocity_block_with_content_list_capacity=0;
+      vbwcl_sizePower = 0;
+      vbwncl_sizePower = 0;
+
+      if (list_with_replace_new) {
+         ::delete list_with_replace_new;
+         list_with_replace_new = 0;
+      }
+      if (list_delete) {
+         ::delete list_delete;
+         list_delete = 0;
+      }
+      if (list_to_replace) {
+         ::delete list_to_replace;
+         list_to_replace = 0;
+      }
+      if (list_with_replace_old) {
+         ::delete list_with_replace_old;
+         list_with_replace_old = 0;
+      }
+      list_with_replace_new_capacity = 0;
+      list_delete_capacity = 0;
+      list_to_replace_capacity = 0;
+      list_with_replace_old_capacity = 0;
+   }
+
+   SpatialCell::SpatialCell(const SpatialCell& other) {
+      std::cerr<<"Warning! Spatial Cell GPU copy constructor called."<<std::endl;
```

Comment:
Should this then crash more thoroughly?

Reply:
No, the copy constructor works, it's just not necessarily optimized. But can update the message to reflect that.